shotgun_api3/shotgun.py

```diff
@@ -97,6 +97,10 @@ class AuthenticationFault(Fault):
     """Exception when the server side reports an error related to authentication"""
     pass
 
+class MultiFactorAuthenticationFault(Fault):
+    """Exception when the server side reports an error related to missing 2FA credentials"""
```

Comment:
This is raised on missing 2FA credentials, right?

Seems like there are some new possible error conditions... do we know what error is raised if:
- 2FA is turned on and the password is wrong but a token is specified?
- 2FA is turned on and the password is right but a wrong token is specified?
- 2FA is turned off and the token is specified?

If the error is only when the 2FA token is needed but missing, then maybe a name like "MissingMultiFactorAuthenticationFault" would be less ambiguous (though longer)?

Also, we should watch out for line length here.